staff/anna-nolla/precourse/bingo/indexPro.js

```diff
@@ -0,0 +1,223 @@
+
+        if (numbers.slice(0, 5).every(x => x === "X") === true && mLine === 0){
+            alert("LINEA !!");
+            mLine = 1;
+        }
+
+        if (numbers.slice(5, 10).every(x => x === "X") === true && mLine === 0){
+            alert("LINEA !!");
+            mLine = 1;
+        }
+
+        if (numbers.slice(10, 15).every(x => x === "X") === true && mLine === 0){
+            alert("LINEA !!");
+            mLine = 1;
+        }
+
+        if (numbers.every(x => x === "X") === true){
+            turnos = j;
+            puntos();
+            player1.points = points;
+            player1.turnos = turnos;
+            alert("BINGO ! ! (en " + turnos + " turnos! ) obtiene " + points + " puntos.");
+            usuario.push(player1);
+            console.log("BINGO ! ! (en " + turnos + " turnos! ) obtiene " + points + " puntos.");
+            do {
+                var newGame = prompt("Would you like to play again ?? Y/N");
+                } 
+            while (newGame != "Y" && newGame != "y" && newGame != "yes" 
+            && newGame != "N" && newGame != "n" && newGame != "no");
+
+            if (newGame == "Y" || newGame == "y" || newGame == "yes"){
+                j = 0;
+                points = 0;
+                getBingo();
+                showBingo();
+                newCard();
+                randomGen();
+                return;
+            }
+
+            else{console.log("Have a nice day ! !");
+            usuario.push(player1);
+            return;}
+        }
+    }
+
+    function newCard() {
+        
+        do {
+            var newBingo = prompt("Would you like new numbers ?? Y/N");
+            } 
+        while (newBingo != "Y" && newBingo != "y" && newBingo != "yes" 
+        && newBingo != "N" && newBingo != "n" && newBingo != "no");
+
+        if (newBingo == "Y" || newBingo == "y" || newBingo == "yes"){
+            getBingo();
+            showBingo();
+            newCard();
+        }
+    }   
+    
+    function randomGen (){
+
+        for (i = 0; i < matchNumber.length; i++){
+            matchNumber[i]= 0;
+        }
+        for (i = 0; i < matchNumber.length; i++){
+            var matchNumb = Math.floor(Math.random() * 35) + 1;
+
+            while (matchNumber.includes(matchNumb) == true){
+                matchNumb =  Math.floor(Math.random() * 35) + 1;
+                }
+            matchNumber[i] = matchNumb;
+        }
+    }
+
+    function getNumber (){
+
+        bingoCard.forEach(function(item){
+            if (item.number == matchNumber[j]){
+                item.number = "X";
+                }
+        })
+        alert("el num es " + matchNumber[j]);
+        
+        do {
+            var getNum = prompt("desea continuear ?? Y/N")
+            } 
+            while (getNum != "Y" && getNum != "y" && getNum != "yes" 
+            && getNum != "N" && getNum != "n" && getNum != "no");

```

Comment:
Here you could use .includes() to avoid too many && conditions
e.g.
var yes = ["Y", "y", "yes"]
while (yes.includes(getNum)

Reply:
changed in all the project.